### PR TITLE
Fixes #10408 - adding checkbox in hosts-index page for keeping selected hosts

### PR DESCRIPTION
**Pull request by amirfefer**

### CODE REVIEW

File: app/assets/javascripts/host_checkbox.js [hunk 1]
```diff
@@ -12,6 +12,10 @@ function hostChecked(box) {
   $.cookie($.cookieName, JSON.stringify($.foremanSelectedHosts), { secure: location.protocol === 'https:' });
   toggle_actions();
   update_counter();
+  if ($.foremanSelectedHosts.length == 0)
+    $("#selected_hosts").hide("slow")
+  else if ($("#selected_hosts").is(":hidden") )
```
Comment (domcleal) [inline app/assets/javascripts/host_checkbox.js]: Use braces for multi-line conditionals.

File: app/assets/javascripts/host_checkbox.js [hunk 2]
```diff
@@ -109,7 +117,8 @@ $(function() {
 });
 
 function submit_modal_form() {
-  removeForemanHostsCookie();
+  if (!document.getElementById("keep_selection").checked)
```
Comment (domcleal) [inline app/assets/javascripts/host_checkbox.js]: jQuery object lookup?

File: app/views/hosts/_list.html.erb [hunk 3]
```diff
@@ -36,6 +36,18 @@
     <% end %>
   </tbody>
 </table>
+<div id="selected_hosts">
+  <table class="table-condensed">
+    <tr>
+      <td class="ca">
+        <%=check_box_tag "keep_selection", "", false, { :title => _("Keep Selected hosts for a future action")} %>
```
Comment (domcleal) [inline app/views/hosts/_list.html.erb]: Missing space after `=`, no need for braces around the hash and "Selected" should be lower case.

File: app/views/hosts/_list.html.erb [hunk 4]
```diff
@@ -36,6 +36,18 @@
...
+      </td>
+      <td>
+        Keep Selected Hosts
```
Comment (domcleal) [inline app/views/hosts/_list.html.erb]: Should be extracted and sentence case.

File: app/views/hosts/_list.html.erb [hunk 5]
```diff
@@ -36,6 +36,18 @@
     <% end %>
   </tbody>
 </table>
+<div id="selected_hosts">
+  <table class="table-condensed">
+    <tr>
+      <td class="ca">
```
Comment (domcleal) [inline app/views/hosts/_list.html.erb]: Is there no helper for this form-type element?

**Comment by domcleal** [comment]
I'm unsure about the placement, it's a long way away from the _Select Action_ dropdown, which is what it's about.

**Comment by amirfefer** [comment]
@domcleal  what about place it in the modal dialog ? 
screenshot from 2015-08-25 14-59-08

**Comment by domcleal** [comment]
That would make more sense, yep.

**Comment by dLobatog** [comment]
Merged as 833dcdeb33d33569f2b1e09e9e7dc6d906590d08, thanks @amirfefer !

**Pull request closed without merging**